### PR TITLE
possible puente removal adjustments

### PR DESCRIPTION
**Pull request by escattone**
@smithellis I collected all of my adjustments for you to consider in this PR. It was easier than entering alll of my suggestions in a review, because I already had made these changes and tested them locally.

### CODE REVIEW

File: kitsune/settings.py [hunk 1]
```diff
@@ -442,8 +442,6 @@ def immutable_file_test(path, url):
             "undefined": "jinja2.Undefined",
             "extensions": [
                 "waffle.jinja.WaffleExtension",
-                "jinja2.ext.autoescape",
-                "jinja2.ext.with_",
```
Comment (escattone) [inline kitsune/settings.py:446]: These are both built into the core of `jinja2` now. They're no longer available from `jinja2.ext`.

File: kitsune/settings.py [hunk 2]
```diff
@@ -426,6 +425,7 @@ def immutable_file_test(path, url):
 
 TEMPLATES = [
     {
+        "NAME": "jinja2",
```
Comment (escattone) [inline kitsune/settings.py:428]: This is necessary for our custom `extract_jinja` function to pick up the `OPTIONS` section. Without this, the `jinja2` extractions will miss a ton of strings.

File: kitsune/dashboards/readouts.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 from django.utils.translation import gettext_lazy as _lazy
 from django.utils.translation import pgettext_lazy
 from django.utils.translation import ugettext as _
-from jinja2 import Markup
+from markupsafe import Markup
```
Comment (escattone) [inline kitsune/dashboards/readouts.py:18]: Both `Markup` and `escape` are no longer available from `jinja2`. They must be imported directly from the `markupsafe` package.

File: kitsune/settings.py [hunk 4]
```diff
@@ -5,7 +5,6 @@
 import os
 import platform
 import re
-from datetime import date
```
Comment (escattone) [inline kitsune/settings.py:8]: Just an unused import I found.

File: kitsune/sumo/management/commands/manage_l10n.py [hunk 5]
```diff
@@ -45,6 +45,8 @@ def handle(self, *args, **options):
                     "locale/templates/LC_MESSAGES/django.pot",
                     "-k",
                     "_lazy",
+                    "-k",
+                    "pgettext_lazy",
```
Comment (escattone) [inline kitsune/sumo/management/commands/manage_l10n.py:49]: Some of the Python files also use `pgettext_lazy`. I'm not sure if it's ever used within JS files.

File: kitsune/settings.py (hunk 1 above)
Comment (smithellis) [reply kitsune/settings.py:446]: Gotcha.

File: kitsune/dashboards/readouts.py (hunk 3 above)
Comment (smithellis) [reply kitsune/dashboards/readouts.py:18]: Yes - hopefully this fixes the build!

File: kitsune/settings.py (hunk 4 above)
Comment (smithellis) [reply kitsune/settings.py:8]: Clean-up never hurts.

**Pull request merged**